src/rpc/client.cpp

```diff
@@ -15,118 +16,245 @@
+    { "zcsamplejoinsplit",           {{}, {}} },
+    { "zcbenchmark",                 {{s, o}, {}} },
+    { "z_getnewaddress",             {{}, {s}} },
+    { "z_getnewaccount",             {{}, {}} },
+    { "z_getaddressforaccount",      {{o}, {o, o}} },
+    { "z_listaccounts",              {{}, {}} },
+    { "z_listaddresses",             {{}, {o}} },
+    { "z_listunifiedreceivers",      {{s}, {}} },
+    { "z_listreceivedbyaddress",     {{s}, {o, o}} },
+    { "z_getbalance",                {{s}, {o, o}} },
+    { "z_getbalanceforviewingkey",   {{s}, {o, o}} },
+    { "z_getbalanceforaccount",      {{o}, {o, o}} },
+    { "z_gettotalbalance",           {{}, {o, o}} },
+    { "z_viewtransaction",           {{s}, {}} },
+    { "z_getoperationresult",        {{}, {o}} },
+    { "z_getoperationstatus",        {{}, {o}} },
+    { "z_sendmany",                  {{s, o}, {o, o, s}} },
+    { "z_setmigration",              {{o}, {}} },
+    { "z_getmigrationstatus",        {{}, {o}} },
+    { "z_shieldcoinbase",            {{s, s}, {o, o}} },
+    { "z_mergetoaddress",            {{o, s}, {o, o, o, s}} },
+    { "z_listoperationids",          {{}, {s}} },
+    { "z_getnotescount",             {{}, {o, o}} },
+    // server
+    { "help",                        {{}, {s}} },
+    { "setlogfilter",                {{s}, {}} },
+    { "stop",                        {{}, {o}} },
 };
 
-set<int> paramsToConvertFor(const string& method) {
+std::string FormatConversionFailure(const std::string& method, const ConversionFailure& failure) {
+    return std::visit(match {
+            [&](const UnknownRPCMethod&) {
+                return tinyformat::format("Unknown RPC method, %s", method);
+            },
+            [&](const WrongNumberOfArguments& err) {
+                return tinyformat::format(
+                        "%s for method, %s. Needed between %u and %u, but received %u",
+                        err.providedArgs < err.requiredParams
+                        ? "Not enough arguments"
+                        : "Too many arguments",
+                        method,
+                        err.requiredParams,
+                        err.requiredParams + err.optionalParams,
+                        err.providedArgs);
+            },
+            [](const UnparseableArgument& err) {
+                return std::string("Error parsing JSON:") + err.unparsedArg;
+            }
+        }, failure);
+}
+
+optional<pair<vector<bool>, vector<bool>>> paramsToConvertFor(const string& method) {
     auto search = rpcCvtTable.find(method);
     if (search != rpcCvtTable.end()) {
         return search->second;
     } else {
-        return set<int>();
+        return nullopt;
     }
 }
 
-/** Non-RFC4627 JSON parser, accepts internal values (such as numbers, true, false, null)
- * as well as objects and arrays.
- */
-UniValue ParseNonRFCJSONValue(const string& strVal)
+optional<UniValue> ParseNonRFCJSONValue(const string& strVal)
```

Comment:
Ditto on adding `std::` prefixes in this commit (and each other commit in turn).